frontend/src/ui/wallet/transfer/transfer-view.js

```diff
@@ -52,28 +67,70 @@ function TransferView() {
         transfer.setMode(tab)
     }
 
+    function getAccountPredefinedDisplayName(address) {
+        if (!window.predefinedAccountDisplayNames) return undefined
+        return window.predefinedAccountDisplayNames[address]
+    }
+
+    async function saveNewAddress() {
+        const {address, ...predefinedAddress} = newAddress
+        predefinedAddress['network'] = network
+        predefinedAddress['federation_address'] = await getFederationAddress(transfer.destination)
+        predefinedAddress.memo.type = transfer.memo?.type || 'none'
```

Comment:
Do not create `predefinedAddress.memo` at all unless `transfer.memo.type` and `transfer.memo.value` set.